re.go

```diff
@@ -23,12 +23,31 @@ custom parsing.
 package re
 
 import (
+	"errors"
 	"fmt"
 	"reflect"
 	"regexp"
 	"strconv"
 )
 
+// Span is a special type designed to be passed via pointer to Scan.  re.Scan
```

Comment:
Nice comment.

I agree with the recommendation for suggesting to capture the entire regexp, but perhaps reword to something like:

This type can be placed anywhere within the list of arguments to scan, but the most typical usage is to find the entire extent of the match, which can be achieved by placing it third (immediately after the regular expression and input), and wrapping the entire regexp in parentheses so that the Span is filled with the extent of the entire match.

Reply:
Done.